packages/web/src/components/gcds-pagination/gcds-pagination.tsx

```diff
@@ -0,0 +1,265 @@
+        :
+          `${I18N[this.lang].previousPage}: ${I18N[this.lang].pageNumberOf.replace('{#}', --page).replace('{total}', this.totalPages).replace('{label}', this.label)}`,
+    }
+
+    if (page == this.currentPage && !end) {
+      linkAttrs['aria-current'] = "page";
+    }
+
+    if (end == "next") {
+      return(<li><a  {...linkAttrs} class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}>{I18N[this.lang].next}<gcds-icon margin-left="spacing-200" name="arrow-right"></gcds-icon></a></li>);
+    }
+    else if (end == "previous") {
+      return(<li><a  {...linkAttrs} class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}><gcds-icon margin-right="spacing-200" name="arrow-left"></gcds-icon>{I18N[this.lang].previous}</a></li>);
+    }
+    else {
+      return (<li class={page != 1 && page != this.totalPages ?  constructClasses(page, this.currentPage, this.totalPages) : "" }><a {...linkAttrs}>{page}</a></li>);
+    }
+
+  }
+
+  /**
+   * Function to render the right steps for display="list" pagination
+   */
+  private configureListPagination() {
+    this.listitems = [];
+    this.mobilePrevNext = [];
+
+    if (this.currentPage != 1) {
+      this.listitems.push(this.configurePaginationStep(this.currentPage, "previous"));
+      this.mobilePrevNext.push(this.configurePaginationStep(this.currentPage, "previous", true));
+    }
+
+    // Flags to see if ellipses already rendered
+    let previousEllipses = false;
+    let nextEllipses = false;
+
+    for (let i = 1; i <= this.totalPages; i++) {
+
+      if (i == 2 && this.currentPage < 6 && this.currentPage > 3 && this.totalPages > 9) {
+        this.listitems.push(<li class={`gcds-pagination-list-mobile-ellipses`} aria-hidden="true"><span class="gcds-pagination-list-ellipses">...</span></li>);
+      } else if (i == 2 && this.totalPages < 10 && this.totalPages > 6 && this.currentPage > 3) {
+        this.listitems.push(<li class={`gcds-pagination-list-mobile-ellipses`} aria-hidden="true"><span class="gcds-pagination-list-ellipses">...</span></li>);
+      }
+
+      if (i == this.currentPage) {
+        this.listitems.push(this.configurePaginationStep(i));
+      } else if (i == 1 || i == this.totalPages) {
+        this.listitems.push(this.configurePaginationStep(i));
+      } else if (i >= (this.currentPage-2) && i <= (this.currentPage+2)) {
+        this.listitems.push(this.configurePaginationStep(i));
+      } else if ((this.currentPage <= 5 && i <= 7) || (this.currentPage >= this.totalPages - 4 && i >= this.totalPages - 6)) {
+        this.listitems.push(this.configurePaginationStep(i));
+      } else if ((this.currentPage == 5 && i == 2) || (this.currentPage == this.totalPages -4 && i == this.totalPages - 1)) {
+        this.listitems.push(this.configurePaginationStep(i));
+      } else if (this.totalPages < 10) {
+        this.listitems.push(this.configurePaginationStep(i));
```

Comment:
Looks like all these if statements up until here push the same configurePaginationStep, could you summarize them as one if statement or condense them a little?